mini-e2e.groovy

```diff
@@ -0,0 +1,69 @@
+def cico_retries = 16
+def cico_retry_interval = 60
+// temporary git repository for testing purpose
+def ci_git_repo = 'https://github.com/Yuggupta27/ceph-csi'
+def ci_git_branch = 'mini-e2e'
+def ref = "master"
+
+node('cico-workspace') {
+	stage('checkout ci repository') {
+		git url: "${ci_git_repo}",
+			branch: "${ci_git_branch}",
+			changelog: false
+	}
+
+	stage('reserve bare-metal machine') {
+		def firstAttempt = true
+		retry(30) {
+			if (!firstAttempt) {
+				sleep(time: 5, unit: "MINUTES")
+			}
+			firstAttempt = false
+			cico = sh(
+				script: "cico node get -f value -c hostname -c comment --retry-count ${cico_retries} --retry-interval ${cico_retry_interval}",
+				returnStdout: true
+			).trim().tokenize(' ')
+			env.CICO_NODE = "${cico[0]}.ci.centos.org"
+			env.CICO_SSID = "${cico[1]}"
+		}
+	}
+
+	try {
+		stage('prepare bare-metal machine') {
+			if (params.ghprbPullId != null) {
+				ref = "master"
+				//ref = "pull/${ghprbPullId}/head"
+			}
+			sh 'scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ./prepare.sh ./single-node-k8s.sh root@${CICO_NODE}:'
+			sh "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./prepare.sh --workdir=/opt/build/go/src/github.com/ceph/ceph-csi --gitrepo=${ci_git_repo} --ref=${ref}"
+		}
+		stage('e2e kube v1.17.5') {
```

Comment:
This will run the stages in sequence. It is fine to only run one e2e in this job, and we can configure multiple jobs with this groovy file with different parameters. Assume that the Kubernetes version gets passed as a parameter (like `params.ghprbPullId`), default to a version of your choice if not set, and only run the e2e for this version.

We can add the parameters when adding the `.yaml` file for the job (under the `jobs/` directory)

Reply:
Thanks, I've made the changes. I'll also add a parameter so that we can reuse the script for different k8s versions.